adapters/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/MessageListenerCodeGenerator.java

```diff
@@ -0,0 +1,253 @@
+                    .addMember(ACTIVATION_CONFIG_PARAMETER, "$L",
+                            generateActivationConfigPropertyAnnotation(SUBSCRIPTION_NAME, subscriptionNameOf(resourceUri, clientId)));
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Generate a single @ActivationConfigProperty annotation to add to the @MessageDriven annotation.
+     * @param name the property name
+     * @param value the property value
+     * @return the annotation specification
+     */
+    private AnnotationSpec generateActivationConfigPropertyAnnotation(final String name, final String value) {
+        return AnnotationSpec.builder(ActivationConfigProperty.class)
+                .addMember("propertyName", "$S", name)
+                .addMember("propertyValue", "$S", value)
+                .build();
+    }
+
+    /**
+     * Convert given URI and component to a camel cased class name
+     *
+     * @param resourceUri URI String to convert
+     * @return camel case class name
+     */
+    private String classNameOf(final ResourceUri resourceUri, final Component component) {
+        return format("%s%s%sJmsListener",
+                capitalize(resourceUri.context()),
+                capitalize(component.pillar()),
+                capitalize(component.tier()));
+    }
+
+    /**
+     * Convert given URI to a valid component.
+     * Takes the last and second to last parts of the URI as the pillar and tier of the component.
+     * @param resourceUri URI of the resource
+     * @param baseUri     base uri of the resource
+     * @return component the value of the pillar and tier parts of the uri
+     */
+    private Component componentOf(final ResourceUri resourceUri, final BaseUri baseUri) {
+
+        final String pillar = resourceUri.pillar();
+        final String tier = resourceUri.tier() != null ? resourceUri.tier() : baseUri.tier();
+        return Component.valueOf(pillar, tier);
+    }
+
+    /**
+     * Parse and format all the message selectors from the Post Action
+     * @param actions Map of ActionType to Action
+     * @return formatted message selector String
+     */
+    private String messageSelectorsFrom(final Map<ActionType, Action> actions) {
+        return format("CPPNAME in('%s')", parse(actions.get(ActionType.POST)));
+    }
+
+    /**
+     * Parse an Action into a message selectors String
+     * @param action Action to parse
+     * @return formatted message selectors String
+     */
+    private String parse(final Action action) {
+        return action.getBody().keySet().stream()
+                .map(this::commandNameOf)
+                .collect(joining("','"));
+    }
+
+    /**
+     * Converts media type String to a command name <p> Command name is equal to everything between
+     * "application/vnd." and the first "+".
+     * @param mediaType String representation of the Media Type
+     * @return command name
+     */
+    private String commandNameOf(final String mediaType) {
```

Comment:
There is a method called fromMediaType in Name, in messaging-core that does this same process.  Can the Name util class version be used instead?